tico/passes/lower_to_resize_nearest_neighbor.py

```diff
@@ -0,0 +1,249 @@
+# Copyright (c) 2025 Samsung Electronics Co., Ltd. All Rights Reserved
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import TYPE_CHECKING
+
+if TYPE_CHECKING:
+    import torch.fx
+from typing import Optional
+
+import torch
+from torch.export import ExportedProgram
+
+from tico.serialize.circle_mapping import extract_shape
+from tico.utils import logging
+from tico.utils.errors import NotYetSupportedError
+from tico.utils.passes import PassBase, PassResult
+from tico.utils.trace_decorators import trace_graph_diff_on_pass
+from tico.utils.validate_args_kwargs import IndexArgs, UpsampleNearest2DVecArgs
+
+
+@trace_graph_diff_on_pass
+class LowerToResizeNearestNeighbor(PassBase):
+    """
+    This pass lowers `aten.index` and `aten.upsample_nearest2d.vec` to `circle_custom.resize_nearest_neighbor` when it is possible.
+
+    Until torch 2.7, `torch.nn.functional.interpolate` is converted to `aten.index` op.
+
+        [EXAMPLE]
+        class InterpolateDouble(torch.nn.Module):
+            def __init__(self):
+                super().__init__()
+
+            def forward(self, x):
+                return torch.nn.functional.interpolate(x, scale_factor=2.0, mode="nearest")
+
+            def get_example_inputs(self):
+                return (torch.randn(1, 2, 3, 4),)
+
+        [EXPORTED GRAPH]
+        [constants]
+            _prop_tensor_constant0 = tensor([0, 0, 1, 1, 2, 2, 3, 3]
+            _prop_tensor_constant1 = tensor([[0], [0], [1], [1], [2], [2]])
+
+        [graph]
+            %_prop_tensor_constant0 : [num_users=1] = placeholder[target=_prop_tensor_constant0]
+            %_prop_tensor_constant1 : [num_users=1] = placeholder[target=_prop_tensor_constant1]
+            %x : [num_users=1] = placeholder[target=x]
+            %_to_copy : [num_users=1] = call_function[target=torch.ops.aten._to_copy.default](args = (%x,), kwargs = {dtype: torch.float32})
+            %index : [num_users=1] = call_function[target=torch.ops.aten.index.Tensor](args = (%_to_copy, [None, None, %_prop_tensor_constant1, %_prop_tensor_constant0]), kwargs = {})
+            %_to_copy_3 : [num_users=1] = call_function[target=torch.ops.aten._to_copy.default](args = (%index,), kwargs = {dtype: torch.float32})
+            return (_to_copy_3,)
+
+        [BEFORE PASS]
+        input - aten.index - output
```

Comment:
```suggestion
        [BEFORE PASS]
        case 1.
        input - aten.index - output

        case 2.
        input - aten.upsample_nearest2d.vec - output
```

It would be nice to add a case for `aten.upsample_nearest2d.vec` :D

Reply:
Let me apply in the next PR :-D